File: BaekJoon/Silver/1874/C++/1874.cpp
#include <cstdio>
#include <vector>
#include <stack>
using namespace std;

int main() {
    int n;
    scanf("%d", &n);
    vector<char> ret;
    stack<int> stk;
    int last_num = 1;
    for(int i = 0; i < n; i++) {
        int num;
        scanf("%d", &num);
        if(num >= last_num) {
            while(last_num <= num) {
                stk.push(last_num);
                ret.push_back('+');
                last_num++;
            }
        }
        if(stk.top() == num) {
            stk.pop();
            ret.push_back('-');
        } else {
            printf("NO\n");
            return 0;
        }
    }
    for(char c : ret)
        printf("%c\n", c);
    return 0;
}

File: BaekJoon/Silver/1874/Python/1874.py
import sys

n = int(sys.stdin.readline())
stack = []
str1 = []
memo = {}
push_num = 1
for i in range(0, n):
    num = int(sys.stdin.readline())
    if num in memo:
        str1 = False
        break
    if len(stack) == 0:
        while push_num <= num:
            stack.append(push_num)
            push_num += 1
            str1.append('+')
        memo[stack.pop()] = True
        str1.append('-')
    elif num <= stack[-1]:
        while len(stack) != 0 and num <= stack[-1]:
            memo[stack.pop()] = True
            str1.append('-')
    else:
        while push_num <= num:
            stack.append(push_num)
            push_num += 1
            str1.append('+')
        memo[stack.pop()] = True
        str1.append('-')

if str1:
    for char in str1:
        print(char)
else:
    print('NO')
